src/tracing/human/mod.rs

```diff
@@ -0,0 +1,188 @@
+use std::fmt;
+use std::fmt::Debug;
+use std::sync::atomic::AtomicBool;
+use std::sync::atomic::Ordering;
+
+use itertools::Itertools;
+use tracing::span;
+use tracing::Event;
+use tracing::Id;
+use tracing::Level;
+use tracing::Subscriber;
+use tracing_subscriber::fmt::format::FmtSpan;
+use tracing_subscriber::fmt::FormattedFields;
+use tracing_subscriber::layer::Context;
+use tracing_subscriber::registry::LookupSpan;
+use tracing_subscriber::Layer;
+
+mod fields;
+use fields::HumanFields;
+
+mod style;
+use style::EventStyle;
+
+mod event;
+use event::HumanEvent;
+use tracing_subscriber::registry::Scope;
+
+#[derive(Debug)]
+pub struct HumanLayer {
+    /// We print blank lines before and after long log messages to help visually separate them.
+    ///
+    /// This becomes an issue if two long log messages are printed one after another.
+    ///
+    /// If this variable is `true`, we skip the blank line before to prevent printing two blank
+    /// lines in a row.
+    ///
+    /// This variable is mutated whenever a [`HumanEvent`] is displayed.
+    last_event_was_long: AtomicBool,
+    /// Which span events to emit.
+    span_events: FmtSpan,
+}
+
+impl Default for HumanLayer {
+    fn default() -> Self {
+        Self {
+            last_event_was_long: Default::default(),
+            span_events: FmtSpan::NONE,
+        }
+    }
+}
+
+impl HumanLayer {
+    pub fn with_span_events(mut self, span_events: FmtSpan) -> Self {
+        self.span_events = span_events;
+        self
+    }
+
+    fn visitor<S>(&self, level: Level, scope: Option<Scope<'_, S>>) -> HumanEvent
+    where
+        S: tracing::Subscriber,
+        S: for<'lookup> LookupSpan<'lookup>,
+    {
+        HumanEvent::new(
+            level,
+            AtomicBool::new(self.last_event_was_long.load(Ordering::SeqCst)),
+            scope,
+        )
+    }
+}
+
+impl<S> Layer<S> for HumanLayer
```

Comment:
I was comparing this to the [`Layer` implementation for `Layer`](https://docs.rs/tracing-subscriber/latest/src/tracing_subscriber/fmt/fmt_layer.rs.html#786-990) and noticed that the other implementation implements thread-safe logging.  Do we also need thread-safe logging here?